src/command.ts

```diff
@@ -48,41 +76,111 @@ export class Command<C extends Context = Context> implements MiddlewareObj<C> {
     return this._languages.get("default")!.description;
   }
 
-  public addToScope(scope: BotCommandGroupsScope, ...middleware: ChatTypeMiddleware<C, "group" | "supergroup">[]): this;
-  public addToScope(scope: BotCommandScopeAllPrivateChats, ...middleware: ChatTypeMiddleware<C, "private">[]): this;
-  public addToScope(scope: BotCommandScope, ...middleware: Array<Middleware<C>>): this;
-  public addToScope(scope: BotCommandScope, ...middleware: Array<Middleware<C>>): this {
+  public addToScope(
+    scope: BotCommandGroupsScope,
+    middleware: MaybeArray<ChatTypeMiddleware<C, "group" | "supergroup">>,
+    options?: Partial<CommandOptions>,
+  ): this;
+  public addToScope(
+    scope: BotCommandScopeAllPrivateChats,
+    middleware: MaybeArray<ChatTypeMiddleware<C, "private">>,
+    options?: Partial<CommandOptions>,
+  ): this;
+  public addToScope(
+    scope: BotCommandScope,
+    middleware: MaybeArray<Middleware<C>>,
+    options?: Partial<CommandOptions>,
+  ): this;
+  public addToScope(
+    scope: BotCommandScope,
+    middleware: MaybeArray<Middleware<C>>,
+    options: Partial<CommandOptions> = this._options,
+  ): this {
```

Comment:
Do I understand it correctly that you no longer want to allow adding several pieces of middleware as several parameters?

Reply:
I couldn't find a way to both make `options` optional and have a rest parameter at the same time. But maybe I'm just missing something.

Reply:
So … you are not happy with the API, but you don't know how to do it better? I don't understand your intention yet

Reply:
I am happy with it. It's just that not allowing multiple middleware as multiple arguments is not really intended, it's a side effect of having an optional parameter.
But passing things as an array is actually fine IMO.